cw-rpm.mk

```diff
@@ -0,0 +1,187 @@
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+# Caller must set the following:
+# PKG_COMPONENT or RPM_COMPONENT (RPM_COMPONENT takes precedence)
+#                      - name of overall component for manifest
+#                        (e.g., sprout)
+# PKG_MAJOR_VERSION or RPM_MAJOR_VERSION (RPM_MAJOR_VERSION takes precedence)
+#                      - major version number for package (e.g., 1.0)
+# PKG_NAMES or RPM_NAMES (RPM_NAMES takes precedence)
+#                      - space-separated base names of packages
+#                        (e.g., sprout sprout-dbg)
+
+# Caller may also set the following:
+# PKG_MINOR_VERSION or RPM_MINOR_VERSION (RPM_MINOR_VERSION takes precedence)
+#                      - minor version number for package (default is current timestamp)
+# REPO_DIR             - path to repository to move packages to (default is unset,
+#                        meaning don't move packages)
+# REPO_SERVER          - username and server to scp pacakges to (default is unset,
+#                        meaning move packages locally)
+# CW_SIGNED            - whether to sign the generated package repository (default
+#                        is unset, meaning don't sign; set to Y to sign).
+#                        IMPORTANT: this signs the repo itself, which means that
+#                        *all* packages in it are marked authentic, not just the
+#                        ones we built just now.
+# HARDENED_REPO_DIR    - path to the hardened repository to move packages to
+#                        (default is unset, meaning don't move packages)
+# HARDENED_REPO_SERVER - username and server for the hardened repository to scp
+#                        packages to (default is unset, meaning move
+#                        packages locally)
+
+# Include common definitions
+include build-infra/cw-pkg.mk
+
+# Default RPM_* from PKG_*
+RPM_COMPONENT ?= $(PKG_COMPONENT)
+RPM_MAJOR_VERSION ?= $(PKG_MAJOR_VERSION)
+RPM_MINOR_VERSION ?= $(PKG_MINOR_VERSION)
+RPM_NAMES ?= $(PKG_NAMES)
+
+RPM_ARCH := $(shell rpmbuild -E %{_arch})
+
+# Commands to build an RPM package repo.
+RPM_BUILD_REPO := createrepo .
+ifeq ($(CW_SIGNED), Y)
+RPM_BUILD_REPO := $(RPM_BUILD_REPO) ; gpg -abs -u $(CW_SIGNER) repodata/repomd.xml
```

Comment:
Does this sign all the packages in the repo or just the XML file. Do we need to do the former?